python/ql/lib/semmle/python/Stmts.qll

```diff
@@ -143,12 +143,46 @@ class Exec extends Exec_ {
   override Stmt getASubStatement() { none() }
 }
 
-/** An except statement (part of a `try` statement), such as `except IOError as err:` */
-class ExceptStmt extends ExceptStmt_ {
-  /* syntax: except Expr [ as Expr ]: */
+/**
+ * An exception handler such as an `except` or an `except*` statement
+ * in a `try` statement.
+ *
+ * We do not wish to expose an abstract class that users are not
+ * meant to extend, so this class is kept private. The user facing
+ * class is `ExceptionHandler`.
+ */
+abstract private class ExceptionHandlerImpl extends Stmt {
+  /** Gets the name of this except group block. */
+  abstract Expr getName();
+
+  /** Gets the type of this except group block. */
+  abstract Expr getType();
+}
+
```

Comment:
When we already know of all the things that should extend this abstract class, I think it would be better to just list them explicitly, and not have an abstract class at all -- at least that's the advice we've written up here: inhttps://github.com/github/codeql/blob/main/docs/ql-design-patterns.md#abstract-classes-as-open-or-closed-unions

```suggestion
```

I guess the `getName`/`getType` can be defined as abstract, such that they **must** be overriden in the subclasses.